src/lib/journey/callbacks/recaptcha/recaptcha.svelte

```diff
@@ -23,57 +38,65 @@
-      if (recaptchaClass.startsWith('h')) {
-        window.hcaptcha.render(recaptchaClass, {
-          sitekey: siteKey,
-          callback: 'frHandleCaptcha',
-          'expired-callback': 'frHandleExpiredCallback',
-          'chalexpired-callback': 'frHandleExpiredCallback',
-          'error-callback': 'frHandleErrorCallback',
-        });
-      } else {
-        window.grecaptcha.render(recaptchaClass, {
-          sitekey: siteKey,
-          callback: window.frHandleCaptcha, // grecaptcha uses different types so passing real function here
-          'expired-callback': window.frHandleExpiredCallback,
-        });
-      }
-    };
-    window.frHandleCaptcha = function handleCaptchaToken(token: string) {
-      callback?.setResult(token);
-    };
-    window.frHandleExpiredCallback = function handleExpiredCallback() {
-      callback?.setResult('');
-      if (recaptchaClass.startsWith('h')) {
-        window.hcaptcha.render(recaptchaClass, {
-          sitekey: siteKey,
-          callback: 'frHandleCaptcha',
-          'expired-callback': 'frHandleExpiredCallback',
-          'chalexpired-callback': 'frHandleExpiredCallback',
-          'error-callback': 'frHandleErrorCallback',
-        });
-      } else {
-        window.grecaptcha.render(recaptchaClass, {
-          sitekey: siteKey,
-          callback: window.frHandleCaptcha, // grecaptcha uses different types so passing real function here
-          'expired-callback': window.frHandleExpiredCallback,
+    if (isV3) {
+      // If ReCaptcha v3, do nothing and return early
+      return;
+    }
+    if (callback) {
+      window.frHandleCaptchaError = handleCaptchaError(callback);
+      window.frHandleCaptcha = handleCaptchaToken(callback);
+      window.frHandleExpiredCallback = function handleExpiredCallback() {
+        callback?.setResult('');
+        renderCaptcha({ nameOfCaptcha: 'hcaptcha', siteKey });
+      };
+      renderCaptcha({
+        nameOfCaptcha: recaptchaClass === 'g-recaptcha' ? 'grecaptcha' : 'hcaptcha',
+        siteKey,
+      });
+    }
+  });
+  // defining this outside of the reactive block and guarding it with a isV3 check so it only runs when v3
+  // is defined as true and we have a recaptcha action to assign.
+  function executeV3Captcha() {
+    if (isV3 && recaptchaAction.length) {
+      try {
+        window.grecaptcha.ready(async function () {
+          const value = await window.grecaptcha.execute(siteKey, {
+            action: recaptchaAction,
+          });
+          callback?.setResult(value);
         });
+      } catch (err) {
+        throw new Error(
+          `Error executing recaptcha. Please make sure you have passed a siteKey and you have loaded the google recaptcha script in your app prior to this Error: ${err}`,
+        );
       }
-    };
+    }
+  }
+  journeyStore.subscribe((value) => {
+    /*
+     * If the recaptcha action name is not passed in
+     * reassign it to the journey name when we have it.
```

Comment:
I feel like this comment is not accurate. If it is accurate, I don't see journey name assignment below. It looks like this journey fallback is happening here: https://github.com/ForgeRock/forgerock-web-login-framework/pull/252/files#diff-8dcfd178b5d5057e0274c71b248321bc4d51d86009561d615a8a0a92a6d4ed3bR208